org.eclipse.gef/src/org/eclipse/gef/ui/palette/FlyoutPaletteComposite.java

```diff
@@ -964,66 +926,58 @@ public void handleEvent(Event event) {
 			final Tracker tracker = new Tracker(Display.getDefault(), SWT.NULL);
 			tracker.setRectangles(new Rectangle[] { origBounds });
 			tracker.setStippled(true);
-			tracker.addListener(SWT.Move, new Listener() {
-				@Override
-				public void handleEvent(final Event evt) {
-					Display.getCurrent().syncExec(new Runnable() {
-						@Override
-						public void run() {
-							Control ctrl = Display.getCurrent().getCursorControl();
-							Point pt = flyout.toControl(evt.x, evt.y);
-							switchDock = isDescendantOf(graphicalControl, ctrl)
-									&& ((dock == PositionConstants.WEST && pt.x > threshold - 10)
-											|| (dock == PositionConstants.EAST && pt.x < threshold + 10));
-							boolean invalid = false;
-							if (!switchDock) {
-								invalid = !isDescendantOf(FlyoutPaletteComposite.this, ctrl);
-							}
-							if (switchDock) {
-								if (dock == PositionConstants.WEST) {
-									threshold = Math.max(threshold, pt.x);
-									threshold = Math.min(threshold, switchThreshold);
-								} else {
-									threshold = Math.min(threshold, pt.x);
-									threshold = Math.max(threshold, switchThreshold);
-								}
-							}
-							Rectangle placeHolder = origBounds;
-							if (switchDock) {
-								if (dock == PositionConstants.EAST) {
-									placeHolder = new Rectangle(0, 0, origBounds.width, origBounds.height);
-								} else {
-									placeHolder = new Rectangle(flyoutBounds.width - origBounds.width, 0,
-											origBounds.width, origBounds.height);
-								}
-								placeHolder = Display.getCurrent().map(flyout, null, placeHolder);
-							}
-							// update the cursor
-							int cursor;
-							if (invalid) {
-								cursor = DragCursors.INVALID;
-							} else if ((!switchDock && dock == PositionConstants.EAST)
-									|| (switchDock && dock == PositionConstants.WEST)) {
-								cursor = DragCursors.RIGHT;
-							} else {
-								cursor = DragCursors.LEFT;
-							}
-							if (isMirrored()) {
-								if (cursor == DragCursors.RIGHT) {
-									cursor = DragCursors.LEFT;
-								} else if (cursor == DragCursors.LEFT) {
-									cursor = DragCursors.RIGHT;
-								}
-							}
-							tracker.setCursor(DragCursors.getCursor(cursor));
-							// update the rectangle only if it has changed
-							if (!tracker.getRectangles()[0].equals(placeHolder)) {
-								tracker.setRectangles(new Rectangle[] { placeHolder });
-							}
-						}
-					});
+			tracker.addListener(SWT.Move, evt -> Display.getCurrent().syncExec(() -> {
```

Comment:
I wonder if the syncExec() is really needed. Aren't SWT events always fired via the UI thread?

Reply:
I think so. I removed it and played with the logic editor's palette and everything where this could be used worked
so I think we are fine.